CI: fix filter rules

psychocoderHPC opened the pull request:
Only CUDA was tested in the CI because the new filter rules from #5255 used the OS version which was not available during the execution of the filter.

psychocoderHPC committed:
CI: fix filter rules

ONly CUDA was tested in the CI because the new filter rules from #5255
used the OS version which was not available during the execution of the
filter.

ci: full-compile

ikbuibui commented on share/ci/n_wise_generator.py:
```diff
@@ -186,7 +186,7 @@ def is_valid_combination(row):
         # clang as host compiler
         if is_clang:
-            if os_name == "ubuntu" and os_version == 22.04 and v_compiler <= 12:
+            if os_name == "ubuntu" and os_version == 22.04 and os_version <= 12:
```
```suggestion
            if os_name == "ubuntu" and os_version == 22.04 and v_compiler <= 12:
```

psychocoderHPC committed:
fix CI issues

Issues become visiable due to reactivation of CI tests for all
compilers.

ci: full-compile

psychocoderHPC committed:
fix g++12 openPMD issue and disable clang 14 tests

- openPMD is using `std::variant` which is producing compile issues with
  g++-12
- disable clang-14 due to compile issues with `std::chrono` from the
  stdlib

ikbuibui reviewed (approved):
As a note : Disables clang 14 for reasons documented in code
